libs/wingc/src/type_check.rs

```diff
@@ -1843,6 +1851,15 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn validate_type_is_optional(&mut self, actual_type: TypeRef, span: &impl Spanned) {
```

Comment:
style: Note sure we need this function, maybe inline it down below do be more explicit (it's only used once).